src/nnbench/types.py

```diff
@@ -110,22 +114,98 @@ class LocalArtifactLoader(ArtifactLoader):
     path : str | os.PathLike[str]
-        The file system pathto the artifact.
+        The file system path to the artifact.
     """
 
-    def __init__(self, path: str | os.PathLike[str]):
+    def __init__(self, path: str | os.PathLike[str]) -> None:
         self._path = path
 
-    def load(self):
+    def load(self) -> Path:
         """
         Returns the path to the artifact on the local file system.
         """
-        return self._path
+        return Path(self._path).resolve()
 
 
-class S3ArtifactLoader(ArtifactLoader):
-    # TODO: Implement this and other common ArtifactLoders here or in a util
-    pass
+class FilePathArtifactLoader(ArtifactLoader):
+    """
+    ArtifactLoader for loading artifacts using fsspec-supported file systems.
+
+    This allows for loading from various file systems like local, S3, GCS, etc.,
+    by using a unified API provided by fsspec.
+
+    Parameters
+    ----------
+    path : str | os.PathLike[str]
+        The path to the artifact, which can include a protocol specifier (like 's3://') for remote access.
+    destination : str | os.PathLike[str] | None
+        The local directory to which remote artifacts will be downloaded. If provided, the model data will be persisted. Otherwise, local artifacts are cleaned.
+    storage_options : dict[str, Any] | None
+        Storage options for remote storage.
+
+    Raises
+    ------
+    ImportError
+        When the fsspec module is not installed.
+    """
+
+    def __init__(
+        self,
+        path: str | os.PathLike[str],
+        destination: str | os.PathLike[str] | None = None,
+        storage_options: dict[str, Any] | None = None,
+    ) -> None:
+        try:
+            from fsspec.core import get_fs_token_paths
+
+            self._get_fs_token_paths = get_fs_token_paths
+
+            from fsspec.implementations.local import LocalFileSystem
+
+            self._LocalFileSystem = LocalFileSystem
+
+            from fsspec.utils import get_protocol
+
+            self._get_protocol = get_protocol
+
+            from fsspec import filesystem
+
+            self._filesystem = filesystem
+        except ImportError:
+            raise ImportError(
+                "The FilePathArtifactLoader depends on fsspec. Do you have it installed?"
+            )
+
+        self.source_path = str(path)
+        if destination:
+            self.target_path = Path(destination)
+            delete = False
+        else:
+            self.target_path = Path(mkdtemp())
+            delete = True
+        self._finalizer = weakref.finalize(self, self._cleanup, delete=delete)
+        self.storage_options = storage_options or {}
+
+    def load(self) -> Path:
+        """
+        Loads the artifact, downloading it if necessary, and returns the local path.
+
+        Returns
+        -------
+        Path
+            The path to the artifact on the local filesystem.
+        """
+        fs, _, _ = self._get_fs_token_paths(self.source_path)
```

Comment:
Since you're ignoring the other two outputs, you could also use `fs = fsspec.filesystem(self.source_path, **self.storage_options)` (or with the options in another set of parentheses, not sure about the calling convention).